assets/src/domain/eventEditor/services/apollo/mutations/datetimes/useDatetimeMutator.ts

```diff
@@ -1,16 +1,61 @@
+import { useCallback, useMemo } from 'react';
+import { useMutation } from '@apollo/react-hooks';
+
 import { CreateDatetimeInput, UpdateDatetimeInput, DeleteDatetimeInput } from './types';
-import { Datetime } from '@edtrServices/apollo/types';
-import { MutatorFnGn } from '@appServices/apollo/mutations';
-import { useMutator, TypeName } from '../';
+import { MutationType } from '@appServices/apollo/mutations';
+import { CREATE_DATETIME, UPDATE_DATETIME, DELETE_DATETIME } from './';
+import useMutationHandler from './useMutationHandler';
+import useUpdateCallback from '../useUpdateCallback';
+import { MutationFunction, TypeName } from '../types';
 
 interface DatetimeMutator {
-	createEntity: MutatorFnGn<CreateDatetimeInput>;
-	updateEntity: MutatorFnGn<UpdateDatetimeInput>;
-	deleteEntity: MutatorFnGn<DeleteDatetimeInput>;
+	createEntity: MutationFunction<any, CreateDatetimeInput>;
+	updateEntity: MutationFunction<any, UpdateDatetimeInput>;
+	deleteEntity: MutationFunction<any, DeleteDatetimeInput>;
 }
 
 const useDatetimeMutator = (id = ''): DatetimeMutator => {
-	return useMutator<Datetime>(TypeName.Datetime, id) as DatetimeMutator;
+	const [createDatetime] = useMutation(CREATE_DATETIME);
+	const [updateDatetime] = useMutation(UPDATE_DATETIME);
+	const [deleteDatetime] = useMutation(DELETE_DATETIME);
+
+	const mutationHandler = useMutationHandler();
+
+	const getUpdateCallback = useUpdateCallback(TypeName.Datetime);
+
+	const createEntity = useCallback(
+		(input) => {
+			const { onUpdate, ...options } = mutationHandler(MutationType.Create, input);
+
+			const update = getUpdateCallback({ onUpdate, mutationType: MutationType.Create, input });
+
+			return createDatetime({ ...options, update });
+		},
+		[createDatetime, getUpdateCallback, mutationHandler]
```

Comment:
we shouldn't have any issues with double memoization here:

![image](https://user-images.githubusercontent.com/1477453/83002283-88e9c300-a015-11ea-9946-dcb718a0b82a.png)

Reply:
the first memoization had happened in `useUpdateCallback`

Reply:
Couldn't really understand what you mean 🤔

Reply:
I was just thinking outloud, my observation was related to the concern that in case of wrong deps array the callback can be cached too aggressively and can cause unwanted results as had happened few weeks ago, but this won't happen because `exhaustive-deps` will statically check each of the callback + the useMemo from the end; all good :+1: